.circleci/config.yml

```diff
@@ -27,28 +27,52 @@ experiment_job: &experiment_job
     EXP: << parameters.exp >>
 
 executors:
-  amphtml-small-executor:
+  docker-small:
     docker:
       - image: cimg/base:stable
     resource_class: small
-  amphtml-medium-executor:
+  docker-medium:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: medium
-  amphtml-large-executor:
+  docker-large:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: large
-  amphtml-xlarge-executor:
+  docker-xlarge:
     docker:
       - image: cimg/node:lts-browsers
     resource_class: xlarge
-  validator-xlarge-executor:
+  jdk-docker-xlarge:
     docker:
       - image: cimg/openjdk:16.0-node
     resource_class: xlarge
+  macos-medium:
+    macos:
+      xcode: 12.4.0
+    resource_class: medium
 
 commands:
+  checkout_repo:
+    steps:
+      - restore_cache:
+          name: 'Restore Git Cache'
+          keys:
+            - git-cache-{{ arch }}-{{ .Branch }}-{{ .Revision }}
+            - git-cache-{{ arch }}-{{ .Branch }}-
+            - git-cache-{{ arch }}-
+      - checkout
+      - save_cache:
+          name: 'Save Git Cache'
+          key: git-cache-{{ arch }}-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - .git
```

Comment:
I deliberately had this code duplicated before between the init job and the downstream jobs: #35475 - TL;DR - saving the cache should only happen once, in the init job. I suggest you keep the code-dup, as it helps clarify which step does what - sometimes offloading things to functions is _less_ readable ;)

Reply:
SGTM